pkg/controller/database/rdsinstance.go

```diff
@@ -158,20 +125,14 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 	cr.Status.AtProvider = rds.GenerateObservation(instance)
 
 	var pw string
-	switch cr.Status.AtProvider.DBInstanceStatus {
-	case v1alpha1.RDSInstanceStateRunning:
-		cr.Status.SetConditions(xpv1.Available())
+	if cr.Status.AtProvider.DBInstanceStatus == v1alpha1.RDSInstanceStateRunning {
 		pw, err = e.createAccountIfNeeded(cr)
 		if err != nil {
 			return managed.ExternalObservation{}, errors.Wrap(err, errCreateAccountFailed)
 		}
-	case v1alpha1.RDSInstanceStateCreating:
-		cr.Status.SetConditions(xpv1.Creating())
-	case v1alpha1.RDSInstanceStateDeleting:
-		cr.Status.SetConditions(xpv1.Deleting())
-	default:
-		cr.Status.SetConditions(xpv1.Unavailable())
 	}
+	// TODO(zzxwill) Without this line, the `Ready` status of `RDSInstance` won't become True
+	cr.Status.SetConditions(xpv1.Available())
```

Comment:
@hasheddan Please help take a look at this TODO, thanks.

Reply:
@zzxwill this is correct, but I am guessing you want to set this (or another condition) based on the switch statement that has been removed above?

Reply:
I also hope `Available` condition is automatically set when the managed resource is up-to-date. But it failed to become `True` without the line.

Oh, I get from the link that `Creating` and `Deleting` can be automatic, but `Available` won't. Got it, Thanks. Let me delete the TODO comment.